libr/parse/p/parse_x86_pseudo.c

```diff
@@ -495,11 +499,45 @@ static bool varsub (RParse *p, RAnalFunction *f, ut64 addr, int oplen, char *dat
 	return ret;
 }
 
+static bool do_replace_segment(ut64 seg, char *data, char *str, int len, const char *needle) {
+	char *ptr, *rest;
+	if ((ptr = strstr (data, needle)) != NULL) {
```

Comment:
the != NULL is unnecessary

Reply:
i prefer not to mix conditionals and assignments, and even less when the variable definition is not set